test/send: don't use SO_ZEROCOPY if not available
Based off https://github.com/axboe/liburing/commit/eb71ed37235d941d5a279c7828090a5ec351fd44
Patch added by Simon Branford (University of Birmingham)
--- test/send-zerocopy.c.orig	2023-02-24 10:37:07.124925000 +0000
+++ test/send-zerocopy.c	2023-02-24 10:40:57.141397866 +0000
@@ -237,11 +237,17 @@
 		}
 	}
 	if (msg_zc) {
+#ifdef SO_ZEROCOPY
 		val = 1;
+
 		if (setsockopt(*sock_client, SOL_SOCKET, SO_ZEROCOPY, &val, sizeof(val))) {
 			perror("setsockopt zc");
 			return 1;
 		}
+#else
+       fprintf(stderr, "no SO_ZEROCOPY\n");
+       return 1;
+#endif
 	}
 	if (tcp) {
 		*sock_server = accept(listen_sock, NULL, NULL);
@@ -401,6 +407,10 @@
 		if (tcp && !client_connect)
 			continue;
 
+#ifndef SO_ZEROCOPY
+       if (msg_zc_set)
+           continue;
+#endif
 		ret = create_socketpair_ip(&addr, &sock_client, &sock_server, ipv6,
 				 client_connect, msg_zc_set, tcp);
 		if (ret) {
